patches/server/0725-make-configured-features-seed-configurable.patch

```diff
@@ -0,0 +1,79 @@
+     private void fixInvulnerableEndCrystalExploit() {
+         fixInvulnerableEndCrystalExploit = getBoolean("unsupported-settings.fix-invulnerable-end-crystal-exploit", fixInvulnerableEndCrystalExploit);
+     }
++
++    public Object2LongOpenHashMap<ResourceLocation> configuredFeatureSeedReplacements = new Object2LongOpenHashMap<>();
++    private void setupConfiguredFeatureSeedReplacer() {
++        configuredFeatureSeedReplacements.defaultReturnValue(-1);
++        String path = "configured-feature.seed-replacements";
++        getLong(path + ".minecraft:disk_clay", -1);
++        ConfigurationSection values = config.getConfigurationSection("world-settings." + worldName + "." + path);
++        if (values == null) {
++            values = config.getConfigurationSection("world-settings.default." + path);
++        }
++        for (String key : values.getKeys(false)) {
++            ResourceLocation resourceKey = new ResourceLocation(key);
++            configuredFeatureSeedReplacements.put(resourceKey, values.getLong(key));
++        }
++    }
+ }
+ 
+diff --git a/src/main/java/net/minecraft/world/level/biome/Biome.java b/src/main/java/net/minecraft/world/level/biome/Biome.java
+index a7a7e6cd87270e64a92448f03f8b0b0c7e375ec7..a8e82019c288c7e44487279a0514db203de839cb 100644
+--- a/src/main/java/net/minecraft/world/level/biome/Biome.java
++++ b/src/main/java/net/minecraft/world/level/biome/Biome.java
+@@ -225,8 +225,10 @@ public final class Biome {
+         Registry<StructureFeature<?>> registry2 = region.registryAccess().registryOrThrow(Registry.STRUCTURE_FEATURE_REGISTRY);
+         int i = GenerationStep.Decoration.values().length;
+ 
++        long populationCopy = populationSeed; // Paper - keep a copy here so that any replacements don't override into features that don't need to be changed
+         for(int j = 0; j < i; ++j) {
+             int k = 0;
++            populationSeed = populationCopy; // Paper also don't want configuredFeatures to pollute StructureFeatures
+             if (structureAccessor.shouldGenerateFeatures()) {
+                 for(StructureFeature<?> structureFeature : this.structuresByStep.getOrDefault(j, Collections.emptyList())) {
+                     random.setFeatureSeed(populationSeed, k, j);
+@@ -263,6 +265,16 @@ public final class Biome {
+                     Supplier<String> supplier3 = () -> {
+                         return registry.getResourceKey(configuredFeature).map(Object::toString).orElseGet(configuredFeature::toString);
+                     };
++                    // Paper start - set the populationSeed back to the one defined by the region and if applicable, replace
++                    populationSeed = populationCopy;
++                    ResourceLocation location = registry.getKey(configuredFeature);
++                    Long replacementSeed = region.getMinecraftWorld().paperConfig.configuredFeatureSeedReplacements.getLong(location);
```

Comment:
Unnecessary boxing. Could use getOrDefault here

Reply:
do you want getOrDefault? getLong uses the default value I set in PaperWorldConfig
` configuredFeatureSeedReplacements.defaultReturnValue(-1);`